Migration.Toolkit.Core/Behaviors/CommandConstraintBehavior.cs

```diff
@@ -0,0 +1,231 @@
+namespace Migration.Toolkit.Core.Behaviors;
+
+using System.Diagnostics;
+using CMS.Base;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Migration.Toolkit.Common;
+using Migration.Toolkit.Core.Abstractions;
+using Migration.Toolkit.Core.MigrationProtocol;
+using Migration.Toolkit.KX13.Context;
+using Migration.Toolkit.KX13.Models;
+
+public class CommandConstraintBehavior<TRequest, TResponse> : IPipelineBehavior<TRequest, TResponse>
+    where TRequest : IRequest<TResponse>
+    where TResponse : CommandResult
+{
+    public const string CmsHotfixVersionKey = "CMSHotfixVersion";
+    public const string CmsHotfixDataVersionKey = "CMSHotfixDataVersion";
+    public const string CmsDbVersionKey = "CMSDBVersion";
+    public const string CmsDataVersionKey = "CMSDataVersion";
+    private readonly ILogger<CommandConstraintBehavior<TRequest, TResponse>> _logger;
+    private readonly IMigrationProtocol _protocol;
+    private readonly IDbContextFactory<KX13Context> _kx13ContextFactory;
+    private readonly ToolkitConfiguration _toolkitConfiguration;
+
+    public CommandConstraintBehavior(
+        ILogger<CommandConstraintBehavior<TRequest, TResponse>> logger,
+        IMigrationProtocol protocol,
+        IDbContextFactory<KX13Context> kx13ContextFactory,
+        ToolkitConfiguration toolkitConfiguration
+    )
+    {
+        _logger = logger;
+        _protocol = protocol;
+        _kx13ContextFactory = kx13ContextFactory;
+        _toolkitConfiguration = toolkitConfiguration;
+    }
+
+    public async Task<TResponse> Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate<TResponse> next)
+    {
+        try
+        {
+            var kx13Context = await _kx13ContextFactory.CreateDbContextAsync(cancellationToken);
+
+            var criticalCheckPassed = PerformChecks(request, kx13Context);
+            if (!criticalCheckPassed)
+            {
+                return (TResponse)(CommandResult)new CommandCheckFailedResult(criticalCheckPassed);
+            }
+        }
+        catch (Exception ex)
+        {
+            _protocol.CommandError<TRequest, TResponse>(ex, request);
+            _logger.LogCritical(ex, "Error occured while checking command constraints");
+            return (TResponse)(CommandResult)new CommandCheckFailedResult(false);
+        }
+        
+        return await next();
+    }
+
+    private bool PerformChecks(TRequest request, KX13Context kx13Context)
+    {
+        var criticalCheckPassed = true;
+        const string supportedVersion = "13.0.64";
+        if (kx13Context.CmsSettingsKeys.FirstOrDefault(s => s.KeyName == CmsDataVersionKey) is { } cmsDataVersion and not { KeyValue: "13.0" })
```

Comment:
Please implement supportedVersion as a composite of major version and hotfix version. And then use these parts in your conditions. Do not use hardcoded values in conditions at several places.